### PR TITLE
Syntax: check Expressions#keyword in to_s

### LINKED ISSUE #5219: 0.24 macro expanding bug
works in 0.23.1

```crystal
macro bla(x)
  {{x}}
end

p bla({:x => 1, :y => (1 + 2), :z => 3})
```

```
Error in 1.cr:7: macro didn't expand to a valid program, it expanded to:

================================================================================
--------------------------------------------------------------------------------
   1.   {:x => 1, :y => 1 + 2
   2. , :z => 3}
--------------------------------------------------------------------------------
Syntax error in expanded macro: bla:2: expecting token '}', not ','

, :z => 3}
^

================================================================================

p bla({:x => 1, :y => (1 + 2), :z => 3})
  ^~~

```

### PR DESCRIPTION
**Pull request by asterite**
Fixes #5219

### CODE REVIEW

File: src/compiler/crystal/syntax/to_s.cr [hunk 1]
```diff
@@ -198,17 +198,24 @@ module Crystal
...
       if @inside_macro > 0
         node.expressions.each &.accept self
       else
-        node.expressions.each do |exp|
+        node.expressions.each_with_index do |exp, i|
           unless exp.nop?
             append_indent
             exp.accept self
-            newline
+            newline unless parens && i == node.expressions.size - 1
```
Comment (MakeNowJust) [inline src/compiler/crystal/syntax/to_s.cr]: It seems also hack 😆

Comment (asterite) [reply src/compiler/crystal/syntax/to_s.cr]: No... when `Expressions#keyword` is `:")"` then there's no need for a final newline.

There are other problems, `accept_with_maybe_begin` should be removed, but we also need `If` to know if it's a regular `if` or `a ? b : c`. 

At least this is a smaller diff and doesn't introduce a named arguments to fix it.

**Comment by asterite** [comment]
Actually, let me remove all hacks from this.

**Comment by MakeNowJust** [comment]
`If#ternary?` is good idea :tada:

**Pull request merged**